os/kernel/task/task_terminate.c

```diff
@@ -160,21 +160,19 @@ int task_terminate(pid_t pid, bool nonblocking)
 	FAR struct tcb_s *dtcb;
 	FAR dq_queue_t *tasklist;
 	irqstate_t flags;
-	trace_begin(TTRACE_TAG_TASK, "task_terminate");
 #ifdef CONFIG_SMP
 	int cpu;
 #endif
-	int ret = OK;
-
 	/* Find for the TCB associated with matching PID */
 
 	dtcb = sched_gettcb(pid);
 	if (!dtcb) {
 		/* This PID does not correspond to any known task */
-		ret = -ESRCH;
-		goto errout_with_lock;
+		return -ESRCH;
 	}
 
+	trace_begin(TTRACE_TAG_TASK, "task_terminate");
```

Comment:
Regardless of this PR, how about removing TTRACE? Because noone uses and maintains this.